pkg/cli/admin/addnodes/create.go

```diff
@@ -0,0 +1,54 @@
+package addnodes
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+func NewCreate(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewCreateOptions(streams)
+	cmd := &cobra.Command{
+		Use:   "create",
```

Comment:
```suggestion
		Use:   "create-image",
```
Later on there would be a create-pxe-files command. This would also line up better with existing openshift-install "agent create image" and "agent create pxe-files" commands.

Reply:
That seems a fair point, I'd like to hear also other comments/suggestion from the others cc @zaneb @pawanpinjarkar @bfournie

Reply:
I was picturing a `--pxe` flag rather than a separate command.
In the installer, `agent create image` and `agent create pxe-files` are separate due to the architecture of the installer preventing us from passing arguments; we had to have separate commands with separate targets. We don't have that restriction here and I think I prefer the flag - it will be easier to document.

Reply:
That said, in the context of the full command `add-nodes create` is a little weird (we're creating... an add-nodes?).
Something like `oc adm new-nodes image` and `oc adm new-nodes add` would make sense to me.

Reply:
> I was picturing a --pxe flag rather than a separate command.

I will also prefer a separate flag

> Something like oc adm new-nodes image and oc adm new-nodes add would make sense to me.

I think it's important to keep the keyword `add` in the command name, so possible options could be:
* `oc adm add-nodes image`
* `oc adm new-nodes add`
(with a slightly stronger preference on the first one, as it seems to describe better what the command does)

Reply:
Looking at the existing` oc adm` commands, I think it does make sense to make the new commands a verb for consistency, i.e. something like `oc adm add-nodes `or `oc adm create-nodes`.  Most (but not all) of the existing commands fit this pattern, e.g
`copy-to-node`
`cordon`
`create-bootstrap-project-template`
`drain`
etc.

Perhaps for the monitor could use `oc adm wait-for-add-nodes` to be consistent with
`wait-for-node-reboot`
`wait-for-stable-cluster`

Reply:
- `oc adm add-nodes-create-image`
- `oc adm add-nodes-create-pxe-files` ( consistent with agent installer)
- `oc adm wait-for-add-nodes` ( consistent with what @bfournie said above and agent installer, day 1 wait-for) , this will monitor the day2 progress

If we plan with `oc adm wait-for-add-nodes` which is actually monitor-add nodes, I wonder if we can rename `monitor-add-nodes` to `wait-for-add-nodes`. Day2 functionality is a DP so maybe we could still make improvements? 

Overall, I would like to have similar names and invocations for the things we are already doing for agent installer , i.e. using a seperate command for pxe like @rwsu said instead of flags, using the same name for monitoring day 2 progress, adding nodes etc.

Reply:
@pawanpinjarkar @bfournie not sure if you missed it but it was intentional to have a common command name (currently `add-nodes`) that will act as a parent for a set of sub-commands (currently `create` and `monitor`). As suggested by the workload team duringt the first meeting, it would be also easier to maintain the code and define the commands ownership (especially for troubleshooting). So the suggested layout should match the form `oc adm <command> <sub-commands>`. For example, based on the previous discussion, a possible layout for the two commands could be:

* `oc adm add-nodes image` - The command that generates the ISO (or pxe) artifacts
* `oc adm add-nodes monitor` - The command that monitors the adding activity

> I wonder if we can rename monitor-add-nodes to wait-for-add-nodes
Sincerely I'm not worried too much about the `node-joiner` tool interface look'n'feel, given that it's an internal tool only

Reply:
> - oc adm add-nodes image - The command that generates the ISO (or pxe) artifacts
> - oc adm add-nodes monitor - The command that monitors the adding activity

This sounds fine to me.

Reply:
Thank you @andfasano. Do we need to stick to `add-nodes`? `oc adm add-nodes image` isn't very readable. We are adding nodes ( and not adding an image)  to expand the cluster so how about 

- `oc adm expand-cluster create-image` - The command that generates the ISO artifacts
- `oc adm expand-cluster create-pxe-files` - The command that generates the PXE artifacts
-  `oc adm expand-cluster monitor-nodes`  or `oc adm expand-cluster wait-for-nodes` - The command that monitors adding nodes to an existing cluster

I still  think `add-nodes` could give users a impression that the command is adding a node to the cluster where as in reality it would be only creating an ISO or PXE files and users need to boot up the nodes by themselves. Happy to talk on 1:1 or team call to discuss further.

Reply:
As a further discussion, I think it would be better to keep the commands in a form that is closer to what they do (they do not directly expand the cluster, rather creating a image that will be used for enlarging it). Considering that `oc` already has a `node-logs` command, considering also the previous bits mentioned by @bfournie, and another [comment](https://issues.redhat.com/browse/WRKLDS-1314) from @zaneb, a reasonable form could be:

* `oc adm node-image create` (with an optional flag for generating the pxe artifacts) for creating the image
* `oc adm node-image monitor` for the monitoring activity

Reply:
(meanwhile, renamed to `node-image` to see how it looks like)

Reply:
Looks like there is a consensus for the latest proposal, resolving the discussion